src/quicklist.c

```diff
@@ -378,6 +378,8 @@ REDIS_STATIC void __quicklistInsertNode(quicklist *quicklist,
         quicklist->head = quicklist->tail = new_node;
     }
 
+    /* Should try to compress new_node, it may be not at head or tail */
+    quicklistCompress(quicklist, new_node);
```

Comment:
In the existing code, ```old_node``` will never be NULL, so ```new_node``` will be compressed in the following ```quicklistCompress(quicklist, old_node); ``` code, so this change is not necessary.